parley/src/layout/line/greedy.rs

```diff
@@ -602,16 +605,50 @@ impl<'a, B: Brush> BreakLines<'a, B> {
                 }
             }
 
+            if quantize {
+                // We round line height to full pixels and then keep track of the overflow.
+                let rounded_line_height = line.metrics.line_height.round();
+                let overflow = line.metrics.line_height - rounded_line_height;
+                line.metrics.line_height = rounded_line_height;
+                // The overflow tracking logic is based on observing Chrome's behavior.
+                // Fractional line height will lead to fluctuating integral perceived line height.
+                if y_overflow <= -0.5 {
+                    // This line will overflow to the previous line.
+                    y -= 1.;
+                    y_overflow += 1.;
+                } else if y_overflow >= 0.5 {
+                    // There will be a 1px gap before the start of the line.
+                    y += 1.;
+                    y_overflow -= 1.;
+                }
+                y_overflow += overflow;
```

Comment:
Maybe I'm missing something but this seems a bit overengineered? When rounding lines to pixel positions in egui, I just accumulated the Y position as a float, and only rounded the Y positions of individual lines. Is this done to make it easier to distribute leading?